src/typechecker.cpp

```diff
@@ -192,4 +200,12 @@ MonoId TypeChecker::mono_unit() {
 	return 4;
 }
 
+MetaTypeId TypeChecker::meta_value() {
+	return 0;
+}
```

Comment:
In the future it might be useful to differentiate mono_values from poly_values (basically just functions). I will go a step further and say this will be useful for the idea of type_mappings (type functions that are not constructors, but proper functions) which you suggested in the code somewhere and I like that idea.

Reply:
Differentiating monos from polys is good for removing a second flag later (is_polimorphic), which appeals greatly to me. Plus it makes sense to me to have all possible type differentiations centralized in the meta type system.

Reply:
I had that same thought, but I ran into some issues:

When you access a poly_value, you always instianciate it first, which turns it into a mono_value. This means that just because you assign an expression into another, the meta type could change, and that does not play nicely with unification.

I "fixed" this by having mono values and poly values be the same thing at the meta type level

Reply:
Ah I see, well that's a mild bummer.